pkg/graph/graph_test.go

```diff
@@ -0,0 +1,122 @@
+package graph
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/go-openapi/swag"
+	. "github.com/onsi/ginkgo/v2/dsl/core"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	cincinnatiPartialResponse = "quay.io/openshift-release-dev/ocp-release@sha256"
+	fakeCincinnatiAddress     = "https://api.example.com/api/upgrades_info/graph"
+)
+
+type ClientMock struct{}
+
+func (c *ClientMock) Do(req *http.Request) (*http.Response, error) {
+	if strings.Contains(req.URL.String(), cincinnatiAddress) {
+		cincinnatiFakeResponse := Response{
+			Nodes: []Release{
+				{
+					Version: "4.13.1",
+					Payload: fmt.Sprintf("%s:foobar1", cincinnatiPartialResponse),
+				},
+				{
+					Version: "4.13.2",
+					Payload: fmt.Sprintf("%s:foobar2", cincinnatiPartialResponse),
+				},
+				{
+					Version: "4.13.3",
+					Payload: fmt.Sprintf("%s:foobar3", cincinnatiPartialResponse),
+				},
+			},
+		}
+
+		responseJSON, err := json.Marshal(cincinnatiFakeResponse)
+		Expect(err).ToNot(HaveOccurred())
+
+		response := &http.Response{
+			StatusCode:    http.StatusOK,
+			Header:        make(http.Header),
+			ContentLength: int64(len(responseJSON)),
+			Body:          io.NopCloser(bytes.NewReader(responseJSON)),
+		}
+
+		return response, nil
+	} else if strings.Contains(req.URL.String(), fakeCincinnatiAddress) {
+		response := &http.Response{
+			StatusCode: http.StatusNotFound,
+			Header:     make(http.Header),
+			Body:       io.NopCloser(bytes.NewReader(nil)),
+		}
+
+		return response, nil
+	}
+
+	return nil, errors.New("test client error, unexpected URL")
+}
+
+var _ = Describe("Test Graph", func() {
+	var (
+		testGraph Graph
+	)
+
+	It("GetReleaseImage - Valid version", func() {
+		version := "4.13.1"
+		graphConfig := GraphConfig{
```

Comment:
nit: we can init the GraphConfig in BeforeEach with common values. Then, on tests, just set for example graphConfig.Version, graphConfig.CincinnatiAddress, etc.

Reply:
Each test has its own `graphConfig`, so  It can't be done using `BeforeEach`